legend-sdlc-server-fs/src/main/java/org/finos/legend/sdlc/server/domain/model/revision/FileSystemRevision.java

```diff
@@ -0,0 +1,106 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.sdlc.server.domain.model.revision;
+
+import org.eclipse.jgit.lib.Ref;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.revwalk.RevWalk;
+import org.finos.legend.sdlc.domain.model.revision.Revision;
+import org.finos.legend.sdlc.server.api.workspace.FileSystemWorkspaceApi;
+import org.finos.legend.sdlc.server.error.LegendSDLCServerException;
+
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.time.Instant;
+
+public class FileSystemRevision implements Revision
+{
+    private final String id;
+    private final String authorName;
+    private final Instant authoredTimestamp;
+    private final String committerName;
+    private final Instant committedTimestamp;
+    private final String message;
+
+    public FileSystemRevision(String revisionId, String authorName, Instant authoredTimeStamp, String committerName, Instant committedTimeStamp, String message)
+    {
+        this.id = revisionId;
+        this.authorName = authorName;
+        this.authoredTimestamp = authoredTimeStamp;
+        this.committerName = committerName;
+        this.committedTimestamp = committedTimeStamp;
+        this.message = message;
+    }
+
+    public static FileSystemRevision getFileSystemRevision(String projectId, String branchName)
+    {
+        try
+        {
+            Repository repo = FileSystemWorkspaceApi.retrieveRepo(projectId);
+            Ref branch = FileSystemWorkspaceApi.getGitBranch(projectId, branchName);
+            RevWalk revWalk = new RevWalk(repo);
+            RevCommit commit = revWalk.parseCommit(branch.getObjectId());
+            revWalk.dispose();
+            String revisionId = commit.getId().getName();
+            String authorName = commit.getAuthorIdent().getName();
+            Instant authoredTimeStamp = commit.getAuthorIdent().getWhenAsInstant();
+            String committerName = commit.getCommitterIdent().getName();
+            Instant committedTimeStamp = commit.getCommitterIdent().getWhenAsInstant();
+            String message = commit.getFullMessage();
+            return new FileSystemRevision(revisionId, authorName, authoredTimeStamp, committerName, committedTimeStamp, message);
+        }
+        catch (IOException e)
+        {
+            throw new LegendSDLCServerException("Repo " + projectId + " can't be found ", Response.Status.NOT_FOUND);
+        }
```

Comment:
Why only catch IOException here? In any case, the caught exception needs to be the cause of the thrown exception. Also, you should probably include the caught exception's message as part of the thrown exception's message.